core/core-awt/src/main/java/org/icepdf/core/pobjects/Destination.java

```diff
@@ -236,6 +240,12 @@ private void parse(List v) {
         Object ob = getDestValue(0, v);
         if (ob instanceof Reference) {
             ref = (Reference) ob;
+        } else if (ob instanceof Integer) {
```

Comment:
Probably suggest somthing like:

// Dest could be a page number instead of a reference
final int idx = (int) ob;
PageTree pageTree = library.getCatalog().getPageTree();
if (idx >= 0 && idx < pageTree.getNumberOfPages()) {
    ref = pageTree.getPageReference(idx);
}

Interstingly this is one of those special corners cases were the notion isn't in the specification.

Reply:
I guess I was tired, I was assuming we didn't have access to library in this object for some reason...       
Thanks for pointing that out.       
Yeah I found that quite strange, I don't know how that pdf got created.

Reply:
No worries and thank you for providing the fix!